sigma/plugins/utility/other/define.py

```diff
@@ -0,0 +1,79 @@
+import discord
+import aiohttp
+from bs4 import BeautifulSoup
+
+async def define(cmd, message, args):
+
+    definition = ""
+
+    defNum = 1
+
+    defLast = False
+    
+    hasPrinted = False
+
+    defURL = r"https://www.merriam-webster.com/dictionary/"
+
+
+    if not args:
+        await message.channel.send(cmd.help())
+        return
+    else:
+        try:
+            defNum = int(args[-1])
+            defLast = True
+        except ValueError:
+            pass
+
+        if not defLast:
+            for i in args:
+                defURL += i
+                defURL += "%20"
+        else:
+            argsCopy = args.copy()
+            del(argsCopy[-1])
+            for i in argsCopy:
+                defURL += i
+                defURL += "%20"
+
+    async with aiohttp.ClientSession() as session:
+        async with session.get(defURL) as resp:
+            rawDef = await resp.text()
+    
+    try:
+        rawDef = rawDef[rawDef.index("<!--JC: actual definitions-->")-10:]
+    except:
+        embed = discord.Embed(title=":unamused: An error has occured. Silly you! Enter a real word next time!", color=0xFF8EE6)
+        await message.channel.send(None, embed=embed)
+        cmd.log("The user did not input a real word. Either that or the Merriam-Webster HTML files were changed, and the whole command needs to be reworked.")
+
+    rawDefArr = rawDef.split("<!--JC: actual definitions-->")
+
+    try:
+        soupObj = BeautifulSoup(rawDefArr[defNum], "html.parser")
+    except:
+        embed = discord.Embed(title=":exclamation: An error has occured. This number definition does not exist; please try again with a different number.", color=0xFF0000)
+        await message.channel.send(None, embed=embed)
+        cmd.log("This error should never trigger, but if it does, something is wrong.")
+
+    for tag in soupObj.find_all(class_="intro-colon"):
+        definition += tag.text
+        if(len(tag.next_sibling.strip()) == 0):
+            if not hasPrinted:
+                for tag2  in soupObj.find_all(class_="sx-link sc"):
+                    definition += tag2.text
+                    definition += " : "
+                hasPrinted = True
+        else:
+            definition += tag.next_sibling
+            definition += " "
+
+    definition.strip()
+    definition.replace(":"," \n— ")
+
+    if not definition:
```

Comment:
I am kinda concerned with how this done. If the website changes is in "sx-link sc" is renamed to something else, then it will always print "That definition doesn't exist, silly". I need the code to differentiate between when the definition is not found and when the web code we are trying to scrape has changed, causing our code not to work on it. That's when I want the cmd.log to print to console an error that needs the bot dev's attention.